lib/services/in-memory-db.service.ts

```diff
@@ -180,6 +286,32 @@ export class InMemoryDBService<T extends InMemoryDBEntity> {
     return this.records.filter(predicate);
   }
 
+  /**
+   * Return an array of records of type `T` filtered with the supplied predicate as an Observable.
+   * Example:
+   * - given records:
+   * ```json5
+   * [
+   *  {
+   *    "id": 1,
+   *    "prop": "test1"
+   *  },
+   *  {
+   *    "id": 2,
+   *    "prop": "test2"
+   *  }
+   * ]
+   * ```
+   * - to find records with a `prop` value of `test1`:
+   * ```ts
+   * const records: Observable<T[]> = service.queryAsync(record => record.prop === 'test1');
+   * ```
+   * @param predicate the filter predicate
+   */
+  public queryAsync(predicate: (record: T) => boolean) {
```

Comment:
Can we add type support to the function: `Observable<T[]>`. Wanting to keep this clear and typed.